src/scanoss/agent/snippet_scan.c

```diff
@@ -0,0 +1,458 @@
+      int i = 0;
+      while(i<licCount){    /*For each item detected... */
+      char * detectedLicense=detectedLicenses[i];   /*... get its name */
+      if(detectedLicenses!=NULL){
+        if(detectedLicense) {
+          
+          int detLic= getLicenseId((unsigned char *) detectedLicense); /* ... from name, get the key of license that matches short_name at license_ref */
+          if(detLic!=0){    /** If the key is valid, insert the result on DB Table */
+            sprintf(auxSql,"INSERT INTO license_file(rf_fk, agent_fk, rf_timestamp, pfile_fk) VALUES(%d,%d, now(), %ld);",detLic,Agent_pk,key);
+            result = PQexec(db_conn,auxSql);
+          } else {/* No license found by its spdx id, lets check if it is a deprecated license */
+            
+            struct node *dep = find(detectedLicense);
+            if(dep) {
+              int oldLic= getLicenseId((unsigned char *)dep->oldName);
+              if(oldLic!=0){ /* It is a deprecated identifier */
+                sprintf(auxSql,"INSERT INTO license_file(rf_fk, agent_fk, rf_timestamp, pfile_fk) VALUES(%d,%d, now(), %ld);",oldLic,Agent_pk,key);
+                result = PQexec(db_conn,auxSql);
+              } 
+              else 
+                {
+                /* It is a deprecated license but it is not included on DB */
+                }
+            }
+          }
+          free(detectedLicenses[i]);
+        }
+        i++;
+      }
+    }
+    }
+    free(contents); 
+   }
+else {
+  LOG_FATAL("Snippet_scan: The scan throws an invalid result");
+  //returned null value. This should never happend
+}
+
+fo_scheduler_heart(1);
+}
+
+/*!
+ * \brief Runs a scan command for a given temp file name
+ * \param path Path to temporary file
+ * \param detected A list of detected license names
+ * \return Licenses detected count.
+ * \note This function allocates memory on each item of "detected" array. Ensure to free after it use,
+ */
+int runScan(char *path,  char **licenses, unsigned char *purl,unsigned char *url, unsigned char *matchType, unsigned char *oss_lines, unsigned char *filePath)
+{
+
+  FILE *Fin;
+  char Cmd[MAXCMD];
+  unsigned char lic[1000]; 
+  memset(Cmd, '\0', MAXCMD);
+  int retLicensesCount = 0;
+  unsigned char apiurl[400];
+  unsigned char key[105];
+ 
+  if(ApiUrl[0]!='\0') 
+  {
+    if(ApiPort[0]!='\0')
+      sprintf((char *)apiurl,"-H %s -p %s",ApiUrl,ApiPort);
+    else
+      sprintf((char *) apiurl,"-H %s",ApiUrl);
+  }  
+  else  
+      memset(apiurl,0,sizeof(apiurl));
+
+  if(accToken[0]!='\0' && accToken[0]!=' ') 
+ {
+    sprintf((char *)key,"-K %s",accToken);
+  }  
+  else  
+        memset(key,0,sizeof(key));
+  ///usr/local/share/fossology/ScanossAgent/agent/./scanner 
+ // sprintf(Cmd, "scanner  %s -o %s.json %s %s", path, path,apiurl,key); /* Create the command to run */
+   sprintf(Cmd, "/usr/local/share/fossology/scanoss/agent/./scanner   %s -o %s.json %s %s", path, path,apiurl,key); /* Create the command to run */
```

Comment:
Very harmful to hardcode the absolute path as it can change during installation.
Would recommend to either use relative path, or use the placeholder as done in other `.in` files.
You can have a look at following file for example and [CMAKE_INSTALL_PREFIX variable](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html):
https://github.com/fossology/fossology/blob/bba08ff923abcb453f1c9d82aeb043cb4ff416bc/install/db/dbmigrate_change_db_encoding.php.in#L61